logtail/logtail.go

```diff
@@ -259,8 +260,16 @@ func (l *logger) uploading(ctx context.Context) {
 
 	for {
 		body := l.drainPending()
+		origlen := -1 // sentinel value: uncompressed
 		if l.zstdEncoder != nil {
-			body = l.zstdEncoder.EncodeAll(body, nil)
+			zbody := l.zstdEncoder.EncodeAll(body, nil)
+			// Only compress if the bandwidth savings are sufficient.
```

Comment:
"Only send it compressed"? The compressing is already done at this point. Should you omit doing the work if it's small enough?

Reply:
Updated comment. Disabled compression entirely for length < 256.